### PR TITLE
solution hw3 seminar 8

### PR DESCRIPTION
**Pull request by neo-pr**
Эдиниэл Де Соуза Морейра
Чтобы запустить код через ГПУ нужно добавить флаг "--device GPU", но не рекомедуется, а то результат почему-то не будет тот же.
По поводу гипер параметров - всего 5 попыток:
1- learning rate = 0.01 (2000 iter/max tree)
2- random search с таким гридои (model - 100 iter/max tree, random search - 10 n_iter)
  a- learning rate: [0.01, 0.3];
  b- l2 leaf reg: [0.001, 0.9]
  c- depth: [5, 6, 7, 8]
который получил результат такой {'depth': 7, 'learning_rate': 0.3, 'l2_leaf_reg': 0.9}
3- {'depth': 7, 'learning_rate': 0.1, 'l2_leaf_reg': 0.1} - 1000 iter/max tree
4- {'depth': 7, 'learning_rate': 0.15, 'l2_leaf_reg': 0.1} - 1000 iter/max tree
5- {'depth': 7, 'learning_rate': 0.1, 'l2_leaf_reg': 0.05} - 1000 iter/max tree (финальные параметры, но там в коде есть)

### CODE REVIEW

**Comment by agcr** [comment]
а вы заливали свой сабмишн на kaggle?

у меня показывает что у вас было 0 попыток сабмишна, и еще почему-то ваши сабмишны были помечены как hidden

**Comment by agcr** [comment]
так, нашел ваш сабмишн в вашем бранче, он выбивает:
Private Score: 0.76007
Public score: 0.76277

т.е. бейзлайн побит

**Comment by agcr** [comment]
принято, 25 баллов

**Comment by neo-pr** [comment]
Спасибо! Да, я залил, но как "late submission". У меня была большая проблема с вторым домашним заданием. Не знал как ускорить. Поэтому не успел раньше залить (до late submission).
